### PR TITLE
Custom nextjs server

### PR DESCRIPTION
**Pull request by eirikurn**
## What

Create `infra-next-server` lib and integrate as custom server in all next.js app.

This is the first stab at this problem. It only works in production and it's a bit boilerplate'y. But it uses only official nx builders (more future-proof?).

It includes most of the same stuff as `infra-nest-server`, but it additionally monkey patches console.* to use winston, since NextJS internals and most React code uses console directly.

Let me know what you think.

## Why

To get better logs, tracing and basic nodejs metrics.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/infra-metrics/src/lib/startMetricServer.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import express from 'express'
-import { register } from 'prom-client'
+import { collectDefaultMetrics, register } from "prom-client";
```
Comment (darrikonn) [inline libs/infra-metrics/src/lib/startMetricServer.ts:2]: needs formatting ;)

**Review by darrikonn — COMMENTED** [review]
Like it!

However, my main concern is that this only works on production (according to the PR's description). Why is that?

Looking at the `serve` target, it has the `build` target as its `buildTarget`; which in turn runs the `build-server` target.
Not sure how `nx` interprets the `:production` suffix .. will only the `:production` commands be run with the `production` flag set, or does it only affect the file replacements and other configurations?

**Comment by eirikurn** [comment]
> However, my main concern is that this only works on production (according to the PR's description). Why is that?
> Looking at the `serve` target, it has the `build` target as its `buildTarget`; which in turn runs the `build-server` target.

Yeah, that's not ideal. The buildTarget is a mistake, it "needs" to point to a `@nrwl/next:build` target to pick up some configurations. Thing is the "next:server" target only supports a simple js custom server.

However, I've recently gotten some tips from @sebastienne on how we can support the custom server locally as well. I'll update this PR soon.

File: scripts/ci/Dockerfile [hunk 2]
```diff
@@ -61,7 +61,7 @@ COPY --from=builder /build/${APP_HOME}/next-modules/ /webapp/next-modules/
 COPY --from=builder /build/${APP_HOME}/public /webapp/public
 COPY --from=builder /build/${APP_DIST_HOME}/.next /webapp/.next
 
-ENTRYPOINT [ "npx", "next", "start", "-p", "4200" ]
+ENTRYPOINT [ "node", "main.js" ]
```
Comment (sindrig) [inline scripts/ci/Dockerfile:64]: it would feel better to me if we would keep being explicit with the port in the Dockerfile

Comment (eirikurn) [reply scripts/ci/Dockerfile:64]: IMO it's more appropriate to configure the port with environment variables. All of our server projects (Next, Nest and Express) support this using the `PORT` variable. At least they should be consistent.

We could add an `ENV PORT=4200` line to this container?

**Comment by eirikurn** [comment]
fyi @darrikonn I've updated the PR with a custom server that works both locally and in docker.

File: scripts/ci/Dockerfile (hunk 2 above)
Comment (sindrig) [reply scripts/ci/Dockerfile:64]: Ah I didn't know that it could be configured via environment. Yes, please add it as a default env to the docker image 👍

**Review by valurefugl — APPROVED** [review]

**Review by darrikonn — DISMISSED** [review]
Nice, good work!

**Review by darrikonn — DISMISSED** [review]

**Review by darrikonn — APPROVED** [review]

**Review by oddsson — APPROVED** [review]
Changes to island-ui things look good to me, the other changes look alien to me so I can't effectively review them 👽

**Pull request merged**